cpp/bench/ann/src/common/dataset.hpp

```diff
@@ -98,6 +102,7 @@ struct dataset {
   [[nodiscard]] auto distance() const -> std::string { return distance_; }
   [[nodiscard]] auto dim() const -> int
```

Comment:
This is [called in the benchmark loop](https://github.com/rapidsai/cuvs/blob/4e0fb09c84c3cd7f8fbd9981bfd78e7013610d10/cpp/bench/ann/src/common/benchmark.hpp#L326). Please adjust the benchmark loop to store this value in a variable, so that we do not need to lock inside the benchmark loop.

Reply:
Thanks for catching this! I updated the logic to cache `dim()` and load it atomically.